magellan-library/src/main/java/com/wealthfront/magellan/navigation/NavigationPropagator.kt

```diff
@@ -0,0 +1,34 @@
+package com.wealthfront.magellan.navigation
+
+import com.wealthfront.magellan.core.Navigable
+
+object NavigationPropagator {
+
+  private var listeners: List<NavigableListener> = mutableListOf()
+
+  fun addNavigableListener(stepNavigationListener: NavigableListener) {
+    listeners = listeners + stepNavigationListener
+  }
+
+  fun removeNavigableListener(navigableListener: NavigableListener) {
+    listeners = listeners - navigableListener
+  }
+
+  fun onNavigate() {
+    listeners.forEach {
+      it.onNavigate()
+    }
+  }
+
+  fun hideCurrentNavigable(currentNavigable: Navigable) {
```

Comment:
will add a lint rule to ensure that `removeNavigableListener` is always called in the same class where its added.

Reply:
Could we add it as a lifecycle-aware component instead?

Reply:
done.